File: Java/17. Letter Combinations of a Phone Number.java
/*
Problem Link: https://leetcode.com/problems/letter-combinations-of-a-phone-number/
*/

import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> letterCombinations(String digits) {
        if (digits.length() == 0) {
            return new ArrayList<String>();
        }

        String[] map = new String[] {
                "0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"
        };

        List<String> res = new ArrayList<String>();

    }
}

File: Problems/17. Letter Combinations of a Phone Number.py
"""
Problem Link: https://leetcode.com/problems/letter-combinations-of-a-phone-number/
"""


class Solution:

    def get_letters(self, digit):
        if digit == "2":
            return ["a", "b", "c"]
        elif digit == "3":
            return ["d", "e", "f"]
        elif digit == "4":
            return ["g", "h", "i"]
        elif digit == "5":
            return ["j", "k", "l"]
        elif digit == "6":
            return ["m", "n", "o"]
        elif digit == "7":
            return ["p", "q", "r", "s"]
        elif digit == "8":
            return ["t", "u", "v"]
        elif digit == "9":
            return ["w", "x", "y", "z"]

    def letterCombinations(self, digits: str) -> List[str]:
        if not digits:
            return []

        if (len(digits) == 1):
            return self.get_letters(digits)

        result = []

        for i in range(len(digits)):
            letters = self.get_letters(digits[i])
            if not result:
                result = letters
            else:
                temp = []
                for j in range(len(result)):
                    for k in range(len(letters)):
                        temp.append(result[j] + letters[k])
                result = temp

        return result
